esx_service/utils/kvESX.py

```diff
@@ -166,35 +168,45 @@ def delete(volpath):
     return True
 
 
+# Align a given string to the specified block boundary.
+def align_str(kv_str, block):
+   # Align string to the next block boundary.
+   new_len = int((len(kv_str) + block - 1) / block) * block - 1
+   padded_str = '{:<{width}}'.format(kv_str, width=new_len)
```

Comment:
The function is returning a string thats ready to be written. With backend like VVOL every write has to be aligned to 4k boundary and also sized to a 4K multiple. Else caller still needs to add a newline and then issue a single IO thats aligned to a 4K IO size and also to a 4K boundary (in this case its always offset 0).